src/db/serde.ts

```diff
@@ -0,0 +1,98 @@
+import { Column, Insertable, Table } from "zapatos/schema";
+
+export interface Hook<U, T> {
+  [t: Table]: { [c: Column]: (x: U) => T };
+}
+
+// TODO: narrow these types
+const DESERIALIZE_HOOK: Hook<any, any> = {};
+const SERIALIZE_HOOK: Hook<any, any> = {};
+
+function applyHook<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  values: Insertable | Insertable[]
+): Insertable | Insertable[] {
+  return Array.isArray(values)
+    ? values.map((v) => applyHookSingle(hook, table, v))
+    : applyHookSingle(hook, table, values);
+}
+
+function applyHookSingle<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  values: Insertable
+): Insertable {
+  const processed: Insertable = {};
+  for (const [k, v] of Object.entries(values)) {
+    const f = hook?.[table]?.[k];
+    processed[k] = f ? f(v) : v;
+  }
+  return processed;
+}
+
+function registerHook<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  column: Column,
+  f: (x: U) => T
+): void {
+  if (!(table in hook)) {
+    hook[table] = {};
+  }
+  hook[table][column] = f;
```

Comment:
do we want to do anything in the case where a hook is already registered? perhaps at least log a warning or something? with static registries like this it's always a concern, though in our case we can be disciplined about configuring all the hooks at one place in the monorepo so it's probably not as big of a concern

Reply:
Filed #6.